CRM/Civisocial/OAuthProvider/Facebook.php

```diff
@@ -173,14 +132,24 @@ public function handleCallback() {
    * @return bool
    */
   public function isAuthorized() {
-    if ($this->token && isset($this->userProfile)) {
+    if ($this->token && !empty($this->userProfile)) {
```

Comment:
either change the name of the method or do the assigning part in a different method. its impossible to find where the userProfile is being assigned currently.